docs/_docs/working-with-source/writing-tests.md

```diff
@@ -0,0 +1,203 @@
+run `./gradlew verifyRoborazziDebug`. This command has been added to the `./dev-scripts/test-all.sh`
+command, so you don't need to run it separately if you run `test-all`.
+
+If `verifyRoborazziDebug` fails it will output comparison images to `<package>/build/outputs/roborazzi`. You can also run `./gradlew compareRoborazziDebug --rerun-tasks` to generate a report of all changed images in `<package>/build/reports/roborazzi`. If you determine that the change is acceptable you can regenerate the screenshots using`recordRoborazziDebug`. (Note that this will regenerate all images from all tests).
+
+More information can be found in [Roborazzi's readme](https://github.com/takahirom/roborazzi#apply-roborazzi-gradle-plugin).
+
+## Testing Figma file rendering
+
+Start with
+the [Compose testing documentation](https://developer.android.com/jetpack/compose/testing) for the
+basics of Compose testing. The below should apply to both unit and integration (AndroidInstrumented)
+tests.
+
+The tests require a `composeTestRule`, though there are two types. The basic one, `createComposeRule()` should be the default choice, as it starts up faster. The alternate is `createAndroidComposeRule<ComponentActivity>()`, for tests (such as ones with Roborazzi) that require access to a running activity.
+
+The DesignDoc to be tested is set via `composeTestRule.setContent()`. For example, HelloWorld's
+basic test uses this line:
+
+```kotlin
+class RenderHelloWorld {
+
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    @Test
+    fun testHello() {
+        composeTestRule.setContent { HelloWorldDoc.mainFrame(name = "Testers!") }
+        ...
+    }
+}
+```
+
+To check that the doc was rendered (meaning was read, deserialized, and displayed): assert that a
+node with the semantic `docIdSemanticsKey` with the value equalling your file's fileId exists. From
+HelloWorld:
+
+```kotlin
+composeTestRule
+    .onNode(SemanticsMatcher.expectValue(docIdSemanticsKey, helloWorldDocId))
+    .assertExists()
+```
+
+### Marking screenshots
+
+Add `.captureRoboImage(<filename>)` to any compose `onNode()` result to take a screenshot there. For
+example, to capture a screenshot of the HelloWorld doc, add this:
+
+```kotlin
+composeTestRule
+    .onNode(SemanticsMatcher.expectValue(docIdSemanticsKey, helloWorldDocId))
+    .captureRoboImage("helloWorld.png")
+```
+
+All tests should output images to `<package>/src/testDebug/roborazzi`. They should also capture an
+image of the last state of the test. These can be accomplished with a `RoborazziRule`,
+see `reference-apps/helloworld/src/testDebug/kotlin/RenderHelloWorld.kt` for the current options
+used. Copy that code block into any tests you create that use Roborazzi, otherwise the screenshots
+will be saved in the default Roborazzi location.
+
+**Remember**, screenshots are **only captured** when running `./gradlew captureRoborazziDebug` and checked
```

Comment:
This failed for me:
* What went wrong:
Task 'captureRoborazziDebug' not found in root project 'DesignCompose' and its subprojects.